src/components/victory-axis.jsx

```diff
@@ -8,116 +8,251 @@ class VictoryAxis extends React.Component {
 
   constructor(props) {
     super(props);
-    const style = this.getStyles();
     this.state = {};
-    this.state.yRange = {min: style.margin, max: style.height - style.margin};
-    this.state.xRange = {min: style.margin, max: style.width - style.margin};
-    this.state.ticks = _.isArray(this.props.ticks) ?
-      this.prop.ticks.length : this.props.ticks;
+    this.state.range = this.getRange();
+    this.state.domain = this.getDomain();
+    this.state.scale = this.setupScale(); // set up a scale with domain and range
+    this.state.ticks = this.getTicks();
+    this.state.tickFormat = this.getTickFormat();
+    this.state.tickSpacing = _.max([this.props.innerTickSize, 0]) + this.props.tickPadding;
   }
 
   getStyles() {
     return _.merge({
-      width: 500,
-      height: 200,
-      margin: 40,
-      xAxis: {
-        stroke: "black",
-        fill: "none",
-        strokeWidth: 0.5,
-        shapeRendering: "crispEdges"
-      },
-      yAxis: {
-        stroke: "black",
-        fill: "none",
-        strokeWidth: 0.5,
-        shapeRendering: "crispEdges"
-      },
-      text: {
-        fontFamily: "sans-serif"
-      }
+      width: "500",
+      height: "200",
+      stroke: "black",
+      fill: "none",
+      strokeWidth: 1,
+      shapeRendering: "crispEdges"
     }, this.props.style);
   }
 
-  getXTransform() {
-    const range = this.state.yRange;
-    return "translate(" + 0 + "," + range.max + ")";
+  getMarginValues() {
+    const minimumMargin = 50;
+    const style = this.getStyles();
+    const top = parseInt(style.marginTop, 10) || minimumMargin;
+    const bottom = parseInt(style.marginBottom, 10) || minimumMargin;
+    const left = parseInt(style.marginLeft, 10) || minimumMargin;
+    const right = parseInt(style.marginRight, 10) || minimumMargin;
+    const orientation = this.props.orientation;
+    return {
+      top: orientation === "top" ? _.max([top, minimumMargin]) : top,
+      bottom: orientation === "bottom" ? _.max([bottom, minimumMargin]) : bottom,
+      left: orientation === "left" ? _.max([left, minimumMargin]) : left,
+      right: orientation === "right" ? _.max([right, minimumMargin]) : right
+    };
+  }
+
+  getDomain() {
+    let domain;
+    if (this.props.domain) {
+      domain = this.props.domain;
+    } else if (this.props.tickValues) {
+      domain = [_.min(this.props.tickValues), _.max(this.props.tickValues)];
+    } else {
+      // we use this.props.scale here, since domain needs to be set on
+      // this.state.scale
+      domain = this.props.scale().domain();
+    }
+    return this.isVertical() ? domain.reverse() : domain;
+  }
+
+  getRange() {
+    const style = this.getStyles();
+    const margin = this.getMarginValues();
+    if (this.isVertical()) {
+      return [margin.top, parseInt(style.height, 10) - margin.bottom];
+    } else {
+      return [margin.left, parseInt(style.width, 10) - margin.right];
+    }
+  }
+
+  isVertical() {
```

Comment:
i see isVertical checked a handful of times in different functions, might be worth storing in state